x-pack/plugins/security_solution/server/lib/risk_engine/configurations.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const riskFieldMap = {
+  '@timestamp': {
+    type: 'date',
+    array: false,
+    required: false,
+  },
+  identifierField: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  identifierValue: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  level: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  totalScore: {
+    type: 'float',
+    array: false,
+    required: false,
+  },
+  totalScoreNormalized: {
+    type: 'float',
+    array: false,
+    required: false,
+  },
+  alertsScore: {
+    type: 'float',
+    array: false,
+    required: false,
+  },
+  otherScore: {
+    type: 'float',
+    array: false,
+    required: false,
+  },
+  riskiestInputs: {
+    type: 'nested',
```

Comment:
I think the one benefit of a `nested` mapping here is that one would be able to find risk scores via a particular `_id/_index` pair. Without `nested` they'd be able to find risk scores via one or the other, but not both. Do you agree with that, and was that your motivation behind this mapping?

Reply:
Yes, you are right about nested.

For this data structure, I think it's not really necessary to be nested, but if we expand it in the future with some other fields for risk reasons, I think we can benefit from the separation which nested give us.

But I remember that you said that nested can be problematic in general, can you give more info on that?

Reply:
> But I remember that you said that nested can be problematic in general, can you give more info on that?

Most of the issues stem from the nested syntax:

* You can't query/aggregate nested fields in the same way as other fields; they need to be [handled differently](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-nested-query.html)
* Sometimes using the incorrect syntax on a nested field will cause a warning/error, but other times it will cause the query to return no results
    * I can come up with specific examples to demonstrate that behavior, but I remember it from when we introduced  the `threat.*` fieldset to kibana alerts.

Reply:
@nkhristinin I overlooked a big issue here: this has no correlation to the existing/planned ECS fields. We need to update these to respect/use ECS.